crates/turbopack-node/src/pool.rs

```diff
@@ -311,35 +294,48 @@ pub struct NodeJsOperation {
     #[allow(dead_code)]
     permit: OwnedSemaphorePermit,
     processes: Arc<Mutex<Vec<NodeJsPoolProcess>>>,
+    allow_process_reuse: bool,
 }
 
 impl NodeJsOperation {
-    fn process_mut(&mut self) -> Result<&mut RunningNodeJsPoolProcess> {
-        self.process
+    async fn with_process<'a, F: Future<Output = Result<T>> + Send + 'a, T>(
+        &'a mut self,
+        f: impl FnOnce(&'a mut RunningNodeJsPoolProcess) -> F,
+    ) -> Result<T> {
+        let process = self
+            .process
             .as_mut()
-            .context("Node.js operation already finished")
+            .context("Node.js operation already finished")?;
+
+        let result = f(process).await;
+        if result.is_err() {
+            self.allow_process_reuse = false;
```

Comment:
Same comment once again, shouldn't we check and throw an error if `with_process` is called when `allow_process_reuse` is false?

Reply:
fixed